R/st_bonds.R

```diff
@@ -0,0 +1,192 @@
+  st_write(data, asset_type = "bonds", destdir = destdir, ..., quiet = quiet)
+}
+
+#' @rdname st_bonds
+#' @export
+st_write_equity <- function(data = st_data_paths(),
+                            destdir = tempdir(),
+                            ...,
+                            quiet = TRUE) {
+  st_write(data, asset_type = "equity", destdir = destdir, ..., quiet = quiet)
+}
+
+#' @rdname st_bonds
+#' @export
+st_write_loans <- function(data = st_data_paths(),
+                           destdir = tempdir(),
+                           ...,
+                           quiet = TRUE) {
+  st_write(data, asset_type = "loans", destdir = destdir, ..., quiet = quiet)
+}
+
+st_df <- function(data, asset_type, ..., quiet = TRUE) {
+  args <- enlist_args(data, asset_type, ..., quiet = quiet)
+
+  args %>%
+    map(~ exec(st, !!!.x)) %>%
+    enframe(value = "st_result") %>%
+    restructure_st_df() %>%
+    unnest(st_result)
+}
+
+st_write <- function(data, asset_type, destdir = tempdir(), ..., quiet = TRUE) {
+  args <- enlist_args(data, asset_type, ..., quiet = quiet)
+
+  for (i in seq_along(args)) {
+    out <- exec(st, !!!args[[i]])
+    out <- unnest(out, .data$st_result)
+    write_csv(out, file = path(destdir, glue("{names(args)[[i]]}.csv")))
+  }
+
+  invisible(data)
+}
+
+enlist_args <- function(data, asset_type, ..., quiet) {
+  dots <- list2(...)
+  long <- keep(dots, ~ length(.x) > 1L) %>%
+    abort_if_no_argument_is_long() %>%
+    abort_if_more_than_one_argument_is_long()
+
+  dots1 <- dots[setdiff(names(dots), names(long))]
+  args1 <- list2(data = data, asset_type = asset_type, !!!dots1, quiet = quiet)
+  nms <- names(long)
+  val <- unlist(long)
+
+  vec_set_names(val, glue("{nms}___{val}")) %>%
+    map(~ append(args1, vec_set_names(.x, nms)))
+}
+
+abort_if_no_argument_is_long <- function(data) {
+  if (identical(length(data), 0L)) {
+    abort(c(
+      "Must privide one argument of `run_stress_test()` with multiple values.",
+      i = "Do you need to use `run_stress_test()` instead?"
+    ))
+  }
+
+  invisible(data)
+}
+
+abort_if_more_than_one_argument_is_long <- function(data) {
+  if (length(data) > 1L) {
+    abort(c(
+      "Must provide no more than one argument with multiple values.",
+      x = glue("Arguments with multiple values: {toString(names(data))}."),
+      i = "Did you forget to pick only one?"
+    ))
+  }
+
+  invisible(data)
+}
+
+restructure_st_df <- function(data) {
+  data %>%
+    separate(.data$name, into = c("arg_name", "arg_value"), sep = "___") %>%
+    unnest(cols = .data$st_result) %>%
+    relocate(.data$st_type, .data$st_name)
+}
+
+st <- function(data, asset_type, ..., quiet = TRUE) {
+  vec_assert(data, character(), size = 2L)
+  data <- vec_set_names(data, envvar_keys())
+  local_envvar(data)
+
+  control_verbosity <- ifelse(quiet, utils::capture.output, identity)
+  invisible(control_verbosity(run_stress_test(asset_type = asset_type, ...)))
+
+  outputs_path() %>%
+    dir_ls() %>%
```

Comment:
robustness: what happens in the, rather likely, case that users randomly drop files in this folder along the way?